tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCoverage.java

```diff
@@ -157,10 +169,245 @@ private String showColumn(Set items) {
     }
 
     public void showDiff(String title, Set<CoreItems> all, Set<CoreItems> coreCoverage) {
-        Set diff = EnumSet.copyOf(all);
+        Set<CoreItems> diff = EnumSet.copyOf(all);
         diff.removeAll(coreCoverage);
         if (diff.size() != 0) {
             errln("\t" + title + ": " + diff);
         }
     }
+
+    static class CoverageStatus {
+
+        private Level level;
+        private boolean inRoot;
+        private boolean inId;
+        private Level languageLevel;
+        private String displayName;
+
+        public CoverageStatus(
+                Level level,
+                boolean inRoot,
+                boolean inId,
+                Level languageLevel,
+                String displayName) {
+            this.level = level;
+            this.inRoot = inRoot;
+            this.inId = inId;
+            this.languageLevel = languageLevel == null ? Level.UNDETERMINED : languageLevel;
+            this.displayName = displayName;
+        }
+
+        @Override
+        public String toString() {
+            return (inRoot ? "root" : "x")
+                    + "\t"
+                    + (inId ? "ids" : "x")
+                    + "\t"
+                    + stringForm(languageLevel)
+                    + "\t"
+                    + stringForm(level)
+                    + "\t"
+                    + displayName;
+        }
+
+        private String stringForm(Level level2) {
+            if (level == null) {
+                return "υnd";
```

Comment:
```suggestion
                level = Level.UNDETERMINED;
```

maybe this should go on the Level object though?

Reply:
Normally we don't need it (a null indicates an anomaly), so probably best to keep just in the test.